app/src/test/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDeviceViewModelTest.kt

```diff
@@ -0,0 +1,148 @@
+import com.wire.kalium.logic.feature.client.RegisterClientUseCase
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.TestCoroutineScheduler
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.amshove.kluent.shouldBe
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeInstanceOf
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+
+@OptIn(ExperimentalMaterialApi::class, ExperimentalCoroutinesApi::class)
+@ExtendWith(CoroutineTestExtension::class)
+class RegisterDeviceViewModelTest {
+
+    @MockK
+    private lateinit var navigationManager: NavigationManager
+    @MockK
+    private lateinit var validatePasswordUseCase: ValidatePasswordUseCase
+    @MockK
+    private lateinit var registerClientUseCase: RegisterClientUseCase
+    @MockK
+    private lateinit var client: Client
+
+    private lateinit var registerDeviceViewModel: RegisterDeviceViewModel
+
+    @BeforeEach
+    fun setup() {
+        MockKAnnotations.init(this)
+        registerDeviceViewModel = RegisterDeviceViewModel(navigationManager, validatePasswordUseCase, registerClientUseCase)
+    }
+
+    @Test
+    fun `when field is empty, button is disabled`() {
+        coEvery { validatePasswordUseCase.invoke(String.EMPTY) } returns false
+        registerDeviceViewModel.onPasswordChange(TextFieldValue(String.EMPTY))
+        registerDeviceViewModel.state.continueEnabled shouldBeEqualTo false
+        registerDeviceViewModel.state.loading shouldBeEqualTo false
+    }
+
+    @Test
+    fun `when field is filled, button is enabled`() {
+        coEvery { validatePasswordUseCase.invoke("abc") } returns true
+        registerDeviceViewModel.onPasswordChange(TextFieldValue("abc"))
+        registerDeviceViewModel.state.continueEnabled shouldBeEqualTo true
+        registerDeviceViewModel.state.loading shouldBeEqualTo false
+    }
+
+    @Test
+    fun `when button is clicked, show loading`() {
+        coEvery { validatePasswordUseCase.invoke(any()) } returns true
+        coEvery { registerClientUseCase.invoke(any(), any(), any()) } returns RegisterClientResult.Success(client)
+
+        registerDeviceViewModel.onPasswordChange(TextFieldValue("abc"))
+        registerDeviceViewModel.state.continueEnabled shouldBeEqualTo true
+        registerDeviceViewModel.state.loading shouldBeEqualTo false
+        registerDeviceViewModel.onContinue()
+        registerDeviceViewModel.state.continueEnabled shouldBeEqualTo false
+        registerDeviceViewModel.state.loading shouldBeEqualTo true
+    }
+
+    @Test
+    fun `when button is clicked and request returns Success, navigateToHomeScreen is called`() {
```

Comment:
We are following given..when...then  pattern for testing
For consistency please use it.
`given button is clicke, when request returns Success, then navigateToHomeScreen is called`

Reply:
Sure, I updated the test names, even the ones that I created earlier and I will keep that in mind. Please let me know it there's still something wrong with the names. 😉